crates/hir-expand/src/attrs.rs

```diff
@@ -346,6 +360,39 @@ impl Attr {
     }
 }
 
+fn unescape(s: &str) -> Option<String> {
+    let mut res = String::with_capacity(s.len());
+    let mut chars = s.chars();
+
+    while let Some(c) = chars.next() {
+        if c == '\\' {
+            match chars.next()? {
+                'n' => res.push('\n'),
+                'r' => res.push('\r'),
+                't' => res.push('\t'),
+                '\\' => res.push('\\'),
+                '\'' => res.push('\''),
+                '"' => res.push('"'),
+                '0' => res.push('\0'),
+                'x' => {
+                    let hex = chars.by_ref().take(2).collect::<String>();
+                    let c = u8::from_str_radix(&hex, 16).ok()?;
+                    res.push(c as char);
+                }
+                'u' => {
+                    let hex = chars.by_ref().take(4).collect::<String>();
+                    let c = u32::from_str_radix(&hex, 16).ok()?;
+                    res.push(char::from_u32(c)?);
+                }
+                _ => return None,
+            }
+        } else {
+            res.push(c);
+        }
+    }
+    Some(res)
+}
+
```

Comment:
shouldve said that before but we have an unescape function we can use from the rustc_lexer, see what we do here https://github.com/rust-lang/rust-analyzer/blob/062e1b9b81542680afde1aa47fa5316ac5133461/crates/syntax/src/ast/token_ext.rs#L215-L250